File: scheduler/core/src/main/java/io/orkes/conductor/dao/scheduler/SchedulerCacheDAO.java
package io.orkes.conductor.dao.scheduler;

import io.orkes.conductor.scheduler.model.WorkflowScheduleModel;

public interface SchedulerCacheDAO {

    void updateSchedule(WorkflowScheduleModel workflowSchedule);

    WorkflowScheduleModel findScheduleByName(String name);

    boolean exists(String name);

    void deleteWorkflowSchedule(String name);

    long getNextRunTimeInEpoch(String scheduleName);

    void setNextRunTimeInEpoch(String scheduleName, long epochMilli);
}

File: scheduler/core/src/main/typescript/io/orkes/conductor/dao/scheduler/SchedulerCacheDAO.ts
import { WorkflowScheduleModel } from 'io/orkes/conductor/scheduler/model/WorkflowScheduleModel';

export interface SchedulerCacheDAO {
    updateSchedule(workflowSchedule: WorkflowScheduleModel): void;

    findScheduleByName(name: string): WorkflowScheduleModel | null;

    exists(name: string): boolean;

    deleteWorkflowSchedule(name: string): void;

    getNextRunTimeInEpoch(scheduleName: string): number;

    setNextRunTimeInEpoch(scheduleName: string, epochMilli: number): void;
}
